File: data/market-local/src/main/java/com/audioquiz/localdatasource/di/DataModule.java
package com.adrian.audioquiz.di.singleton.database;

import android.content.Context;

import com.adrian.api.data.datasources.session.SessionLocal;
import com.adrian.database.cache.SessionCache;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import dagger.hilt.InstallIn;
import dagger.hilt.android.qualifiers.ApplicationContext;
import dagger.hilt.components.SingletonComponent;


@Module
@InstallIn(SingletonComponent.class)
public class DataModule {

    @Singleton
    @Provides
    SessionLocal providesSessionLocal(@ApplicationContext Context context) {
        return new SessionCache(context);
    }

}

File: data/market-repository/src/main/java/com/audioquiz/data/di/DataModule.kt
package com.audioquiz.data.di

import dagger.Binds
import dagger.Module
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import com.audioquiz.data.repository.MarketRepositoryImpl
import com.audioquiz.domain.repository.MarketRepository

@Module
@InstallIn(SingletonComponent::class)
interface DataModule {

    @Binds
    fun bindMarketRepository(
        marketsRepository: MarketRepositoryImpl,
    ): MarketRepository
}
